### ISSUE TITLE
Add Policy levers

### ISSUE THREAD

**Issue by toddmedema** [toddmedema/electrify#63]
Perhaps all policies are sliders, where the middle means nothing, and going farther on either side costs more and has marginally less and less impact (although some policies could be either a boolean checkbox, or one-sided slider, e.g. Time of Use Rates)

Specific policy ideas:
1. Subsidize heat pumps, or make permitting more complex - costs money to subsidize, but increased adoption reduces peak energy load from heating in winter
2. Subsidize solar, or make permitting more complex - costs money to subsidize, reduces consumer demand based on solar availability
3. Time of Use Rates - slightly pushes consumer demand to hours that you specify 
4. Demand Response technologies - costs money to implement; the more cumulative money you spend, the larger % of demand/timing you control (would need some new UI work for managing the Demand Response once implemented)
5. Block data centers, eg https://github.com/toddmedema/electrify/issues/183
6. TODO: Brainstorm what other levers utilities have here - subsidizing energy efficiency upgrades?

**Comment by toddmedema** [comment]
## Revised implementation plan: small choices, visible consequences

Reviewed September 7, 2026 against the current Insights controls, demand model, simulation clock, finance/forecast paths, save/replay boundaries, and #183. This replaces the previous implementation proposal. The issue's original ideas remain the longer-term backlog; the first release is intentionally smaller.

**Recommendation:** introduce customer programs in Insights, with two initial choices—efficiency rebates and rooftop solar rebates. Let players experiment with one program at a time, see the cost and demand shape change, and apply it next month. Add time-of-use pricing only after that loop works. Keep demand-response scheduling, heating conversion, and data-center approvals out of the initial release.

The learning goal is: **“I can help meet electricity needs by changing demand, but each approach helps at different times and has costs.”** The fun comes from recognizing a problem, trying a choice, and seeing whether it helps the grid the player built.

### Critique of the previous plan

| Previous proposal | Problem for players or implementation | Revision |
| --- | --- | --- |
| Six levers, signed sliders, two schedules, investment stocks, and one package | Too many new concepts before the first useful decision; changing several things makes cause and effect harder to understand. | Two programs initially; edit and apply one at a time. No package builder. |
| A slider from obstruction to subsidy for most policies | Implies permissions, funding, and adoption are opposite ends of the same physical quantity. Arbitrary precision adds work without necessarily adding strategy. | Explicit `Off / Small / Large` funding choices. Keep approvals separate from subsidies. |
| Lock everything for 12 months | Makes mistakes costly to explore and introduces a second calendar rule. Existing scenarios include one-, six-, and twelve-month runs. | Changes apply next month; funding persists until changed. No blanket annual lock or renewal chores. Slow adoption supplies the long-term consequence. |
| “Heat pumps reduce winter peaks” | Not a generally valid electricity lesson. Replacing resistance heating can reduce electricity use; replacing fossil heating adds electric load. The current aggregate temperature curve does not identify heating technology. | Defer, or later explicitly model a named replacement cohort. Never implement a generic cold-weather demand discount labeled heat pumps. |
| Per-control demand effects and annual cost | Does not show whether a program helps the actual shortage or whether lost sales outweigh avoided operating cost. Annual labels alone also obscure the next charge. | Preview the demand curve and program spending, with net utility cash impact available in details; use matching comparison horizons. |
| One shared symmetric response curve for every lever | Diminishing returns make sense for rebate spending, but not automatically for tariffs or yes/no connection decisions. | Program-specific, bounded effects. For rebates, Small should be more cost-effective per extra unit than Large; Large buys more absolute impact. |
| Enactment triggers one reforecast | Necessary, but does not define draft preview, pending edits, cancellation, month boundaries, or forecast-state isolation. | Specify those transitions and reuse the real forecast on isolated copies. |
| Data-center blocking as a general slider | Could remove the core challenge of Data Center Boom with little downside. Existing scheduled loads are already authored scenario commitments. | Separate future connection decisions; require explicit policy-sensitive projects and objectives/tradeoffs before enabling them. |

Keep the strong parts: integration with Insights, one clear primary action, shared pure calculations, deterministic simulation, explicit finance accounting, read-only replay, and responsive/accessibility checks.

### 1. Player flow

**Find it.** Keep the current rate slider and its immediate behavior. Add a quiet **Customer programs** entry beside it within the existing planning-control area; avoid hiding the familiar rate behind a new modal. When active, summarize `1 program active · $…/month`; when a change is pending, show `Change starts Oct 2026`. Do not add a top-level navigation destination or crowd the overview with another dashboard.

**Choose a program.** Open a compact desktop dialog and a full-screen phone view. Show two short rows, each with its current funding and one plain-language explanation:

- **Efficiency rebates:** “Help homes and businesses use less electricity.”
- **Rooftop solar rebates:** “Help customers make electricity during daylight.”

Selecting a row opens that program's detail. Keep only one program expanded. Use `Off / Small / Large` labeled choices, with their actual monthly cost; no percentage slider or annual investment input. Off means no new funded adoption, not removal of upgrades already installed.

**Try it.** Changing a choice creates a local draft. Show one short mechanism sentence, one tradeoff, and a compact demand comparison. Example solar copy: “Grows gradually. Helps in daylight; does not directly cover an evening peak.” Avoid claiming that lower demand always means higher profit or lower emissions.

**Apply it.** Use one primary button: **Apply next month**. Show the actual month beside it and explain that charges start then. Use **Cancel** as the secondary action; omit a redundant Reset button. Disable Apply for an unchanged or invalid draft. Applying closes the detail and returns to the programs list with a clear pending label. Do not require a second confirmation for an ordinary program change.

**Notice the result.** At activation, add one concise event-log entry. Selecting **View demand** from the program opens the existing demand view; do not force-open charts or show recurring congratulations. Keep an unobtrusive status such as “Building up” or “Fully adopted” in details. The game must remain enjoyable with every program Off.

### 2. Timing and interaction rules

- A funding change takes effect at the next simulation month boundary. One detailed game day represents a month: use the existing calendar helpers, and show calendar dates in the UI.
- Current funding stays active until that boundary. The latest accepted edit to the same program replaces its pending change. **Cancel scheduled change** removes only that pending change; it does not stop an active program.
- Funding continues monthly until changed. Setting Off stops new funded adoption and spending from its effective month. Previously installed efficiency and solar remain; document the initial model's within-run persistence assumption rather than silently erasing them.
- Adoption advances once per month using a bounded, saturation-aware rule. Charge only for enrollment/upgrades the program actually funds, up to the displayed monthly budget. Show the amount as **up to $…/month**, make estimates distinguish budget from expected spending, and stop new spending when no eligible adoption remains. Avoid infinite benefit or indefinite charges for a completed rollout.
- Pause live play while the decision dialog is open, using the app's pause conventions; restore the previous speed only if the dialog still owns that pause. Cancel/close discards the local draft. Disable background game shortcuts while the dialog owns focus.
- Revalidate at dispatch, including replay mode, scenario availability, current date, allowed tier, and pending state. An end-of-run activation must not silently accept a change that can never take effect.
- Keep programs unavailable in the existing short introductory tutorials initially. Enable only in scenarios with enough time to see adoption; historical and institutional availability belongs in authored data, not an arbitrary unlock currency or menu full of disabled future controls.

### 3. A preview that teaches one thing at a time

Use one small before/after demand chart for a clearly named representative month, initially the first effective month. Directly label **Current plan** and **With this change**, with different line styles as well as color. Show energy purchased from the utility, not total societal energy use. A text summary must communicate the same result without requiring chart interpretation.

Below the chart show only:

1. **Program spending** for that month, beneath the monthly budget choice.
2. **Peak demand** before → after in scenario-appropriate units, calculated as each curve's maximum. Do not report reduction at the old peak as though it were reduction of the new peak.

A secondary **After 12 months** comparison reveals gradual adoption, with an actual month/year; omit or shorten it when beyond scenario end. First-month and later effects must be distinguishable, and tiny initial changes should read “Little change yet,” not rounded zero without explanation.

Collapsed **Why this changes** details contain energy supplied, change in utility cash over the selected horizon, and the relevant tradeoff. Cash must include program spending, lost sales, and dispatch-dependent costs. Explain “less electricity sold” when relevant. Warn inline if projected cash becomes negative; follow the game's existing debt rules rather than treating a forecast warning as an automatic spending prohibition.

The comparison holds weather, fuel prices, fleet, customer assumptions, and other accepted/pending programs constant, changing only the selected draft. Label values as estimates conditional on those assumptions. Do not add invented confidence intervals or a “best policy” recommendation score. Do not sum separately calculated program benefits: interacting programs need a combined calculation.

Run a cancellable/debounced preview after a choice settles, on isolated forecast state. Keep Apply disabled while the current draft's preview is incomplete or failed, and suppress stale results. Do not synchronously regenerate long-range forecasts during pointer movement. Enactment still produces one authoritative reforecast.

### 4. Initial simulation scope

**Efficiency:** gradually reduce eligible residential/commercial grid consumption, bounded by remaining adoption potential. Preserve other demand categories. The tradeoff is program cost and reduced electricity sales in exchange for avoided generation costs and possible reliability benefits. Start with a clearly described simplified end-use reduction; do not pretend it is a technology-resolved building model.

**Rooftop solar:** build persistent behind-the-meter capacity through rebates. Derive output from each tick's existing irradiance calculation. Subtract from eligible residential/commercial demand after efficiency; cap at their remaining load. In the initial model, surplus is curtailed, with no customer export payment or utility-owned generation credit. Explain this simplification in details. Utility generation/emissions respond through actual dispatch, not a separate automatic “carbon saved” multiplier.

Both are optional choices, not upgrades everyone should maximize. Balance Small, Large, and Off against cash constraints, fleet mix, and the timing of shortages. Prefer transparent authored caps/costs to a forced quadratic formula. Costs and eligible populations must scale sensibly between municipal and national games and use existing inflation conventions.

### 5. Explicit follow-ups, not launch dependencies

| Lever | Smallest understandable later version | Must resolve before shipping |
| --- | --- | --- |
| Time-of-use pricing | Off plus a small set of named peak-window presets, with visible hours and a before/after curve. | A price incentive must have defined billing/revenue and customer-response treatment; the current game bills supplied kWh at one rate. If using a simplified revenue-neutral tariff, state that assumption. Shift only a capped flexible share, conserve modeled daily energy, and show any new rebound peak. No free peak removal. |
| Demand response | A customer flexibility program with a simple automatic rule before any manual scheduler. | Participation caps, event limits, compensation, and rebound; differentiate direct control/incentives from time-of-use pricing. Do not count the same flexible load twice. Model-only energy conservation applies to the chosen shifting design, not all real-world demand response. |
| Heat-pump rebates | A named cohort, such as replacing electric-resistance heaters. | An explicit eligible heating load and cold-weather performance; fossil-to-electric conversion needs a different demand/emissions treatment. |
| Data-center connections | An occasional project choice, e.g. accept or defer, with MW, date, and revenue consequences. | Reuse #183's `loadAdditions`; only explicitly uncommitted projects may change. Persist decisions and prevent retroactive removal of operating load. Keep objectives achievable and avoid bypassing authored challenges. Generic regional-growth controls are separate later work. |

No policy tree, package editor, arbitrary hourly scheduler, signed permitting slider, new satisfaction meter, or compulsory policy tutorial in the first release. Do not invent a new penalty system just to make these choices look balanced.

### 6. Engineering sequence

1. **Small domain surface.** Add typed program IDs/tiers, current funding, pending changes with effective month, and persistent adoption in `src/Types.tsx` and the game slice. Put availability, caps, costs, and copy in `src/data/Policies.tsx`; put pure transition/cost/demand helpers in `src/helpers/Policies.tsx`. Do not build speculative state for deferred levers.
2. **Explicit actions.** Add validated per-program schedule/cancel actions rather than policy objects through `delta`. Record each accepted user action once in replay; the deterministic month rollover activates it without recording another user action. Log activation once in live play, never from a forecast. Keep cross-slice actions behind `GameActions.tsx` where needed.
3. **Monthly transition and forecast parity.** Advance activation, funded adoption, and costs in a documented order shared by live and forecast paths. Repeated forecasting must never advance real stocks or charge real cash. `generateNewTimeline` and `reforecastSupply` currently copy selected mutable state, so explicitly isolate new nested policy state. A forecast must project future adoption on its own state; “only advance stocks on real rollovers” must not accidentally freeze forecast benefits.
4. **Demand integration.** Apply effects after `demandByTypeAt` in `getDemandW`, then derive total demand from the resulting categories. Keep the all-Off/no-stock path bit-for-bit identical to today's demand calculation. Preserve existing authored additions, weather, customer growth, and import-order boundaries. Track solar effects as reduced utility demand, not both demand reduction and supplied generation.
5. **Complete accounting.** Add `expensesPolicy` to live/forecast ticks, `HistoryForecastShared`, all zero/default constructors, monthly history aggregation, derived expense/profit calculations, Finances, and applicable debrief calculations. In particular, update `src/helpers/DateTime.tsx`; adding a Finances row alone does not update totals. Use the existing representative-day/year scaling so a monthly budget is not charged once per tick or multiplied twice.
6. **Thin responsive UI.** Extract only the necessary planning summary, program list/detail, and preview components from Insights. Reuse MUI/theme/chart systems and the existing three-destination navigation; do not broadly rewrite the workbench. Add concise Manual explanations and a contextual hint within the program details.
7. **Compatibility.** Normalize old saves to Off/no stocks/no pending changes and missing historical policy expenses to zero. Validate malformed policy state in imported saves/replays. Retain current save/replay versions only with demonstrated compatibility; otherwise deliberately version and define migration/rejection behavior. New-action replay compatibility with older clients must not be implied by neutral-state equivalence.

### 7. Acceptance and validation

**Learning/usability gate:** test the two-program slice with a few first-time players before adding more. Without opening the Manual, they should be able to find programs, describe the spending and start date, apply/cancel a change, and explain why rooftop solar does not directly solve an evening shortage. They should distinguish a lower peak from lower total energy, and understand that stopping funding does not remove installed upgrades. Treat observed confusion as a reason to simplify copy/controls, not add a tutorial wall. This is a proposed playtest gate, not a claim that testing has happened.

**Meaningful automated coverage:**

- Neutral saves and action streams reproduce existing outcomes; policies with persistent stock continue to work after funding stops and after save/load.
- Schedule, replace, cancel, activation, and scenario-end boundaries; rejected/no-op actions do not create charges or spurious events.
- Monthly spending is bounded by budget and eligible adoption, stops at saturation, and matches cash, history, projections, and Finances totals.
- Solar has no night effect; adoption saturates; demand remains non-negative; category totals equal demand; combined efficiency/solar cannot double-count savings.
- Preview, long-range forecast, and live run agree under identical assumptions across month boundaries and forecast resolutions; previews do not mutate live state, logs, or replay.
- Same recorded actions replay to the same result; playback and reducer guards reject interactive policy edits.
- Existing import-order test remains green. Add economic invariants to the real headless simulator rather than duplicating game formulas.

**UI checks:** keyboard-operated funding choices; dialog focus/return focus and Escape; no background shortcut actions; concise settled-result announcements; chart text alternative; draft cancellation and pending replacement; stale/error preview handling. Verify 320 px, typical phone, 1024 px panes, and wide desktop in light/dark mode. Use 4 px spacing, 40/44 px targets, text alongside state colors, and a primary action that remains reachable without covering content.

**Balance:** compare Off/Small/Large and combined programs in at least a cash-constrained run, a daytime-heavy load case, and an evening/winter shortage case. Confirm there is no universal maximize-everything answer and no requirement to use programs to win existing scenarios. Keep the Data Center Boom commitments unchanged in this release.

For implementation, use Node 24 with `npm ci`; finish with `npm run check`, `npm run build`, and responsive/tutorial `npm run test:e2e` coverage appropriate to the changed UI. Deliver the complete two-program slice with accounting, previews, compatibility, and validation before adding time-of-use pricing. Deferred controls should not exist as empty launch UI.

### Educational references

The heating distinction above is supported by DOE's explanation of [heat pumps replacing resistance heating](https://www.energy.gov/policy/articles/heat-pumps-keep-homes-warm-and-bills-low-winter) and its research summary on [winter peak growth from heating electrification](https://eesm.science.energy.gov/publications/dual-impacts-space-heating-electrification-and-climate-change-increase-uncertainties). EIA distinguishes [time-of-use customer response and utility-controlled demand response](https://www.eia.gov/todayinEnergy/detail.php?id=650). These inform the educational framing; numerical game balance still needs explicit authored assumptions and playtesting.

<details>
<summary>Earlier visual explorations — superseded, not implementation specifications</summary>

The earlier mocks illustrate the former six-lever package concept. They have not been revised to match this narrower flow and should not determine launch scope or interaction details.

- [Earlier desktop mock](https://github.com/user-attachments/assets/f78b8eab-7ddb-48dc-9e7c-9371b97bb507)
- [Earlier mobile mock](https://github.com/user-attachments/assets/839f3857-370e-40e5-90d0-71a1f1a4fc8e)

</details>
